source/ceylon/net/http/server/MatcherRule.ceylon

```diff
@@ -16,6 +17,22 @@ shared abstract class Matcher() {
             => Or(this, other);
 }
 
+"Matcher to leverage Undertow's template mechanism for path templates and path parameters.
+ It should be given to an Endpoint without combining it with other matchers.
+ 
```

Comment:
Providing an api that allows you to combine matchers in unworkable combination doesn't seems the right way. Lets see what king of api we want to provide first, I see this two options:

a) all matchers can be combined, which allows to define a rule like:
path startsWith('/rest') or matches('/rest/a/{b}/c/{d}')
which seems a valid use case, but the same result can be achieved by creating two endpoints.

b) We have a set of matchers that can be used and not combined Matcher|TemplateMatcher. Where name like SimpleMatcher would fit better in place of Matcher.

Thinking again I believe we should go for option A. While it doesn't have much benefits, we are not loseing anything if we support it, right?

It looks like our issue to support option "A" is in our backend implementation. Do we need to use Undertows path handler, do we lose anything if we create request path parser in CeyonRequestHandler? Maybe use Undertows path parser instead of whole Handler.